### PR TITLE
added support for different interpolations on the same chart

### PR DESCRIPTION
**Pull request by revlucio**
We needed multiple different interpolations on the same chart, and after reading through #193 I found @LostInBrittany had a fork that solved the issue but there wasn't a pull request, so I'm submitting this now.

### CODE REVIEW

**Comment by LostInBrittany** [comment]
Thanks for doing it! My fork solved it in a quick and dirty way, without
unit tests or anything, that was why I didn't submit a PR, I want
particularly proud of it.

**Comment by revlucio** [comment]
Um... I didn't write any unit tests either. Seems to work fine though! Our clients are definitely impressed.

**Comment by LostInBrittany** [comment]
It has worked for me since I coded it in my fork, but I thought it was poor
form to send the PR without doing
all that. Next time I'll be less shy and I'll push it :)

File: src/models/line.js [hunk 1]
```diff
@@ -145,23 +145,29 @@ nv.models.line = function() {
                 .data(function(d) { return [d.values] });
 
             linePaths.enter().append('path')
+              .each(function (d, i) {
+                d3.select(this)
                 .attr('class', 'nv-line')
                 .attr('d',
                     d3.svg.line()
-                    .interpolate(interpolate)
+                    .interpolate(interpolate[d[0].series])
```
Comment (liquidpele) [inline src/models/line.js]: I'm confused, where is the "interpolate" array coming from?  It's still defined above as just "linear" ?

**Comment by liquidpele** [comment]
This is probably a good case for adding a special parameter on each series to control the interpolation.   We've been trying to document those here:

https://github.com/novus/nvd3/wiki/Data-Attributes

**Comment by liquidpele** [comment]
Closing this PR as abandoned.. .

**Comment by revlucio** [comment]
I forgot all about this. We've been using the fix in production for months. I might try and clean it up and submit a better PR.

**Pull request closed without merging**